app/forms/image-upload.tsx

```diff
@@ -211,7 +212,12 @@ export function CreateImageSideModalForm() {
 
   const createDisk = useApiMutation('diskCreate')
   const startImport = useApiMutation('diskBulkWriteImportStart')
-  const uploadChunk = useApiMutation('diskBulkWriteImport')
+  const uploadChunk = useApiMutation(
+    'diskBulkWriteImport',
+    {},
+    // use both the general abort signal for the whole process and a per-request timeout
+    { signal: anySignal([AbortSignal.timeout(30000), abortController.current?.signal]) }
+  )
```

Comment:
TODO: While this seems to work, I'm pretty sure it should not, and it should be pretty easy to break with multiple hangs, because this is using the same timeout signal for all requests. Every mutate call should get its own.